### PR TITLE
Upgrade to Go 1.15

### PR DESCRIPTION
**Pull request by mrnugget**

### CODE REVIEW

**Comment by Strum355** [comment]
Was reminded of the following issue, want to raise awareness to it for those that mightnt have seen it before. Havent checked it out recently so I'm not sure why it has been closed: https://github.com/golang/go/issues/39568

**Comment by mrnugget** [comment]
> Was reminded of the following issue, want to raise awareness to it for those that mightnt have seen it before. Havent checked it out recently so I'm not sure why it has been closed: [golang/go#39568](https://github.com/golang/go/issues/39568)

Thanks! It sounds to me like it can be fixed by creating new certificates, but I'd like @sourcegraph/distribution to chime in here.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/sourcegraph/sourcegraph
 
-go 1.14
+go 1.15
```
Comment (keegancsmith) [inline go.mod:3]: FYI this is just the minimum version we allow when running the go command. We don't have to bump this every go release. I don't see a reason to bump it now. If you do want to increase the minimum, you should also bump `internal/version/minversion`.

**Review by keegancsmith — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (mrnugget) [reply go.mod]: Good point, changed it back.

**Comment by pecigonzalo** [comment]
@mrnugget Nice! I see some images using `1.13` could update those to the same version?
- Prometheus
- Grafana

**Comment by mrnugget** [comment]
> Nice! I see some images using `1.13` could update those to the same version?

Done!

**Comment by pecigonzalo** [comment]
Reading the thread there, this seems to have been fixed by AWS.
Check: 
- https://github.com/golang/go/issues/39568#issuecomment-665863345
- https://github.com/golang/go/issues/39568#issuecomment-671424481

I would add a note to the changelog as this requires customers to update their environments if using an external DB. AWS already added a note and fix on their side https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL-certificate-rotation.html

**Comment by mrnugget** [comment]
Thank you! I added a comment to the changelog file that contains reference to the relevant issues and docs.

File: CHANGELOG.md [hunk 2]
```diff
@@ -27,6 +27,8 @@ All notable changes to Sourcegraph are documented in this file.
...
 - The experimental feature flag `andOrQuery` is deprecated. [#13435](https://github.com/sourcegraph/sourcegraph/pull/13435)
+- Negated content search is now also supported for unindexed repositories. Previously it was only supported for indexed repositories. Negated content search requires setting `"search.migrateParser": true` in settings and is only supported for literal and regexp queries.
+- Sourcegraph is now built with Go 1.15.0. We recommend that customers using Sourcegraph with an external database and and connecting to it using SSL/TLS check whether the certificate is up to date. See these comments for more information: https://github.com/golang/go/issues/39568#issuecomment-665863345 and https://github.com/golang/go/issues/39568#issuecomment-671424481. And [AWS' updated documentation on updating the SSL/TLS certificate](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL-certificate-rotation.html). [#13433](https://github.com/sourcegraph/sourcegraph/pull/13433)
```
Comment (pecigonzalo) [inline CHANGELOG.md:31]: Im not sure its completely clear, what do you think of:

```suggestion
- Sourcegraph is now built with Go 1.15.0.
  - Go `1.15.0` introduced changes to SSL/TLS connection validation which requires certificates to include a `SAN`  which is not included in older certificates which relied on the `CN` field. You might see an error like `x509: certificate relies on legacy Common Name field`. We recommend that customers using Sourcegraph with an external database and and connecting to it using SSL/TLS check whether the certificate is up to date. 
  - RDS Customers please reference [AWS' documentation on updating the SSL/TLS certificate](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL-certificate-rotation.html). 
  - For additional information check [#13433](https://github.com/sourcegraph/sourcegraph/pull/13433)
```

**Comment by slimsag** [comment]
This basically means upon upgrade to 3.20 Sourcegraph will immediately be broken for some customers - I don't think a CHANGELOG entry is enough here. I think we need at minimum:

1. A warning at the top of the CHANGELOG section
2. A warning in our blog post
3. A note in the marketing email that gets sent out

Ask in #marketing or #product for how to handle the last two.

It would be ideal if we could catch/detect that exact error case and link to a docs page that explains it.

**Comment by mrnugget** [comment]
Makes sense. But since I've opened this PR with the goal to get rid of VS Code telling me my Go version has changed and estimated the effort from that point of view, I'm putting this on hold until I have more time.

**Comment by mrnugget** [comment]
Closing in favor of #18026

**Pull request closed without merging**